File: src/main/java/com/nickcoblentz/montoya/settings/ByteExtensionSetting.java
package com.nickcoblentz.montoya.settings;

import burp.api.montoya.MontoyaApi;

public class ByteExtensionSetting extends GenericExtensionSetting<Byte> {

    public ByteExtensionSetting(MontoyaApi api, String name, String storageKey, Byte defaultValue, ExtensionSettingSaveLocation defaultSaveLocation) {
        super(api, name, storageKey, defaultValue, defaultSaveLocation);
    }

    @Override
    public void loadFromProject() {
        setCurrentValue(getApi().persistence().extensionData().getByte(getStorageKey()));
    }

    @Override
    public void loadFromPreferences() {
        setCurrentValue(getApi().persistence().preferences().getByte(getStorageKey()));
    }

    @Override
    public String toString() {
        return String.format("%02X",getCurrentValue());
    }

    @Override
    public Byte parseString(String value) {
        return Byte.valueOf(value);
    }

    public void setCurrentValue(String value) {
        if(value!=null)
            setCurrentValue(parseString(value));
    }
}

File: src/main/java/com/nickcoblentz/montoya/settings/ByteExtensionSetting.kt
package com.nickcoblentz.montoya.settings

import burp.api.montoya.MontoyaApi

class ByteExtensionSetting(
    api: MontoyaApi,
    name: String,
    storageKey: String,
    defaultValue: Byte,
    defaultSaveLocation: ExtensionSettingSaveLocation
) :
    GenericExtensionSetting<Byte>(api, name, storageKey, defaultValue, defaultSaveLocation) {
    override fun loadFromProject() {
        currentValue = api.persistence().extensionData().getByte(storageKey)
    }

    override fun loadFromPreferences() {
        currentValue = api.persistence().preferences().getByte(storageKey)
    }

    override fun toString(): String {
        return String.format("%02X", currentValue)
    }

    override fun saveToProject() {
        api.persistence().extensionData().setByte(storageKey, currentValue)
    }

    override fun saveToPreferences() {
        api.persistence().preferences().setByte(storageKey, currentValue)
    }

    override fun parseString(value: String): Byte {
        return value.toByte()
    }

    override fun setCurrentValue(value: String?) {
        if (value != null) currentValue = parseString(value)
    }
}